### ISSUE TITLE
An error occurred while printing "\r" by print!

### ISSUE THREAD

**Issue by captain-kuan** [rust-lang/rust#96055]
I tried this code:

```rust
    print!("{}", String::from("123456\r"));
    print!("{}", String::from("hello"));
```

I expected to see this happen: *explanation*
123456
hello

Instead, this happened: *explanation*
No errors are reported, but output:
hello6
### Meta

`rustc --version --verbose`:
```
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: x86_64-pc-windows-msvc
release: 1.59.0
LLVM version: 13.0.0
```

<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by jyn514** [comment]
Newlines are `\r\n`, not `\r`. `\r` means "go to the start of the line" and will overwrite anything previously written.

**Comment by captain-kuan** [comment]
my bad

**Comment by captain-kuan** [comment]
> I tried this code:
> 
> ```rust
>     print!("{}", String::from("123456\r"));
>     print!("{}", String::from("hello"));
> ```
> 
> I expected to see this happen: _explanation_ 123456 hello
> 
> Instead, this happened: _explanation_ No errors are reported, but output: hello6
> 
> ### Meta
> `rustc --version --verbose`:
> 
> ```
> rustc 1.59.0 (9d1b2106e 2022-02-23)
> binary: rustc
> commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
> commit-date: 2022-02-23
> host: x86_64-pc-windows-msvc
> release: 1.59.0
> LLVM version: 13.0.0
> ```
> 
> Backtrace



> Newlines are `\r\n` on windows, not `\r`. `\r` means "go to the start of the line" and will overwrite anything previously written.

thx,my bad
